spring-rabbit/src/main/java/org/springframework/amqp/rabbit/log4j2/AmqpAppender.java

```diff
@@ -169,8 +173,9 @@ public static AmqpAppender createAppender(
 			theLayout = PatternLayout.createDefaultLayout();
 		}
 		AmqpManager manager = new AmqpManager(configuration.getLoggerContext(), name);
+		manager.uri = uri;
 		manager.host = host;
-		manager.port = port;
+		Optional.ofNullable(port).ifPresent(v -> manager.port = Integers.parseInt(v));
```

Comment:
Well, is it really expectable that we are fine to fall back to `0` if our string is empty?
```
public static int parseInt(final String s, final int defaultValue) {
        return Strings.isEmpty(s) ? defaultValue : Integer.parseInt(s);
    }

    public static int parseInt(final String s) {
        return parseInt(s, 0);
    }
```
I mean maybe it would be better to use `if (StringUtils.hasText())` instead of this `Optional.ofNullable()` and don't use that `Integers` from log4j2 already...
Although that might be a fact that log4j2 XML config parser treats empty attributes (e.g. `port=""`) as nulls anyway...